graphqlmetrics/server.go

```diff
@@ -0,0 +1,97 @@
+package graphqlmetrics
+
+import (
+	"context"
+	"errors"
+	"github.com/wundergraph/cosmo/graphqlmetrics/gen/proto/wg/cosmo/graphqlmetrics/v1/graphqlmetricsv1connect"
+	"go.uber.org/zap"
+	brotli "go.withmatt.com/connect-brotli"
+	"golang.org/x/net/http2"
+	"golang.org/x/net/http2/h2c"
+	"net/http"
+	"time"
+)
+
+// Option defines the method to customize Server.
+type Option func(s *Server)
+
+type Server struct {
+	server     *http.Server
+	listenAddr string
+	logger     *zap.Logger
+
+	metricsService graphqlmetricsv1connect.GraphQLMetricsServiceHandler
+}
+
+func NewServer(metricsService graphqlmetricsv1connect.GraphQLMetricsServiceHandler, opts ...Option) *Server {
+	s := &Server{
+		metricsService: metricsService,
+		listenAddr:     ":4005",
+		logger:         zap.NewNop(),
+	}
+
+	for _, opt := range opts {
+		opt(s)
+	}
+
+	s.bootstrap()
+
+	return s
+}
+
+func (s *Server) bootstrap() {
+	mux := http.NewServeMux()
+	path, handler := graphqlmetricsv1connect.NewGraphQLMetricsServiceHandler(
+		s.metricsService,
+		// Compress responses with Brotli.
+		brotli.WithCompression(),
```

Comment:
Does it make a difference? We're not sending back much, right?

Reply:
This is required to decompress brotli. The comment is misleading.

Reply:
Fixed